controllers/resource/resource_generator.go

```diff
@@ -106,9 +106,47 @@ func (r *ResourceGenerator) GetDoguDeployment(doguResource *k8sv1.Dogu, dogu *co
 		return nil, wrapControllerReferenceError(err)
 	}
 
+	applyValuesFromCustomDeployment(deployment, customDeployment)
+
 	return deployment, nil
 }
 
+func applyValuesFromCustomDeployment(desiredDeployment *appsv1.Deployment, patchingDeployment *appsv1.Deployment) {
+	logger := ctrl.Log
+
+	if patchingDeployment != nil {
+		if patchingDeployment.Spec.Template.Spec.ServiceAccountName != "" {
+			logger.Info("Found service account in custom deployment from k8s folder. Injecting into the deployment generated by the operator.")
+			desiredDeployment.Spec.Template.Spec.ServiceAccountName = patchingDeployment.Spec.Template.Spec.ServiceAccountName
+		}
+
+		for i, containerGenerated := range desiredDeployment.Spec.Template.Spec.Containers {
+			for _, containerProvided := range patchingDeployment.Spec.Template.Spec.Containers {
+				if containerGenerated.Name == containerProvided.Name {
+					if containerProvided.VolumeMounts != nil && len(containerProvided.VolumeMounts) > 0 {
+						logger.Info("Found custom volume mounts in custom deployment from k8s folder. Injecting into the deployment generated by the operator.")
+						if containerGenerated.VolumeMounts == nil {
+							containerGenerated.VolumeMounts = []corev1.VolumeMount{}
+						}
+
+						containerGenerated.VolumeMounts = append(containerGenerated.VolumeMounts, containerProvided.VolumeMounts...)
+						desiredDeployment.Spec.Template.Spec.Containers[i] = containerGenerated
```

Comment:
Just set the mounts: `desiredDeployment.Spec.Template.Spec.Containers[i].VolumeMounts = containerGenerated.VolumeMounts`